superprofile/initFaunaOnce.sh

```diff
@@ -17,20 +20,33 @@ sleep $waittime
 DBF="log/init_key_flag"
 container_name="faunadb"
 healthy="healthy"
-container_state="$( docker inspect -f '{{ .State.Health.Status }}' ${container_name} )"
-
-if [ "$container_state" != $healthy ]; then
-    echo "Docker container needs extra startup time, please increase the \$waittlist value in initFaunaOnce.sh"
-    echo "Process ended with health status of Container: $container_state"
-else
-    docker exec -it faunadb /bin/sh  /var/log/faunadb/initialize.sh $1
-    if [ -f log/dbkey ] && [ ! -f log/init_key_flag ]; then
-        echo "Copying over secrets to ../app/.env"
-        printf '\nNEXT_PUBLIC_FAUNA_SECRET=' | cat - ./log/dbkey >> ../app/.env &&
-        printf '\nNEXT_PUBLIC_FAUNA_DOMAIN'="http://localhost:8084/graphql" >> ../app/.env
-        touch $DBF &&
-        echo "-- All set, superprofile launch 🚀"
+
+check_and_start_fauna_container() {
+    container_state="$( docker inspect -f '{{ .State.Health.Status }}' ${container_name} )"
+
+    if [ "$container_state" != $healthy ]; then
```

Comment:
I think it would be better if we check for `container_health==starting` since it will not be efficient to go on increasing wait-time for the container if the health becomes `unhealthy`.

```
health_status = "unhealthy"

// current logic
if (health_status != "healthy") {
             // evaluates to true
             increase_waittime() //unnecessary wait
 }

// updated logic
if (health_status == "starting") {
             // evaluates to false
             // do not run the below steps and quit
            increase_waittime()
 }
```

The health status can be one of the following:
**starting**  – Initial status when the container is still starting
**healthy**   – If the command succeeds then the container is healthy
**unhealthy** – If a single run of the  takes longer than the specified 
  timeout then it is considered unhealthy. If a health check fails then the 
  will run retries number of times and will be declared unhealthy 
  if the  still fails.
  
Thank you!

Reply:
Ohhh!! That's a really good catch!⚡️ Thank you so much @Dnouv for the detailed explanation and review, I'll make the necessary changes!

Thank you 😄!